src/attribute_set.rs

```diff
@@ -180,6 +180,14 @@ macro_rules! evdev_enum {
     };
     ($t:ty, $($(#[$attr:meta])* $c:ident = $val:expr,)*) => {
         impl $t {
+            pub fn from_str(s: &str) -> Option<$t> {
```

Comment:
Could you make this a `std::str::FromStr` impl instead of just an associated function? You could probably just have a `struct EnumParseError(());` error type.

Reply:
Good point. 

I've updated the PR.